src/Contrib/Jaeger/HttpSender.php

```diff
@@ -22,75 +26,105 @@ class HttpSender
 
     private TProtocol $protocol;
 
+    private BatchAdapterFactoryInterface $batchAdapterFactory;
+
     public function __construct(
         ClientInterface $client,
         RequestFactoryInterface $requestFactory,
         StreamFactoryInterface $streamFactory,
         string $serviceName,
-        ParsedEndpointUrl $parsedEndpoint
+        ParsedEndpointUrl $parsedEndpoint,
+        BatchAdapterFactoryInterface $batchAdapterFactory = null
     ) {
         $this->serviceName = $serviceName;
 
-        $transport = (new ThriftHttpTransport(
-            $client,
-            $requestFactory,
-            $streamFactory,
-            $parsedEndpoint
-        ));
+        $this->protocol = new TBinaryProtocol(
+            new ThriftHttpTransport(
+                $client,
+                $requestFactory,
+                $streamFactory,
+                $parsedEndpoint
+            )
+        );
+
+        $this->batchAdapterFactory = $batchAdapterFactory ?? new BatchAdapterFactory();
+    }
+
+    public function send(iterable $spans): void
+    {
+        $spansGroupedByResource = self::groupSpansByResource($spans);
+        $batchArray = $this->createBatchesPerResource($spansGroupedByResource);
```

Comment:
Can you please make this one call and use "batches"?
```php
 $batches = $this->createBatchesPerResource(
    self::groupSpansByResource($spans)
 );
```

Reply:
Yup done